### ISSUE TITLE
`flatten` should support `Datatype::FixedSizeList`

### ISSUE THREAD

**Issue by universalmind303** [apache/datafusion#9138]
### Describe the bug

flatten only supports `List` not `FixedSizeList`

### To Reproduce

```sql
select flatten(some_fixed_size_list);

Error: Internal error: Not reachable, data_type should be List.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker

```

### Expected behavior

flatten behaves the same on both `List` and `FixedSizeList`

### Additional context

```rs
 BuiltinScalarFunction::Flatten => {
                fn get_base_type(data_type: &DataType) -> Result<DataType> {
                    match data_type {
                        DataType::List(field) => match field.data_type() {
                            DataType::List(_) => get_base_type(field.data_type()),
                            _ => Ok(data_type.to_owned()),
                        },
                        _ => internal_err!("Not reachable, data_type should be List"),
                    }
                }

                let data_type = get_base_type(&input_expr_types[0])?;
                Ok(data_type)
            }
```

probably just need to add a branch here for `FixedSizeList`

**Comment by Weijun-H** [comment]
This issue can be fixed  by #9108

**Comment by alamb** [comment]
> This issue can be fixed by #9108

Thanks @Weijun-H  -- I will put #9108 on my review queue

**Comment by jonahgao** [comment]
Close it because it is already supported.

```sh
DataFusion CLI v38.0.0

> select flatten(arrow_cast(make_array([[1], [2]], [[3], [4]]), 'FixedSizeList(2, List(List(Int64)))')) as a;
+--------------+
| a            |
+--------------+
| [1, 2, 3, 4] |
+--------------+
1 row(s) fetched.
Elapsed 0.011 seconds.
```
